server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -264,21 +267,26 @@ public void close() {
      * completed
      * @throws InterruptedException If the current thread is interrupted
      */
-    public synchronized boolean awaitClose(long timeout, TimeUnit unit) throws InterruptedException {
-        if (closed) {
-            return true;
-        }
-        closed = true;
+    public boolean awaitClose(long timeout, TimeUnit unit) throws InterruptedException {
+        try {
+            lock.lock();
```

Comment:
I did some quick searching here because Im just not too familiar w/ this stuff, and it looks like most of the time the `lock.lock()` is outside of the try, which i assume is to throw the `InterruptedException`, rather than to execute the `lock.unlock()` before throwing. Should the `lock.lock()` be in the `try` ?

Reply:
The `lock.lock()` should be outside of the try. If the lock is not obtained then calling unlock will throw an `IllegalMonitorStateException` since the thread does not hold the lock.

Reply:
ty for the explain!!

Reply:
thanks @baz and @jaymode - fixed in https://github.com/elastic/elasticsearch/pull/41451/commits/a9e0f3bd8db3b7aa79ba2166895057c198475f56